caliban/config/__init__.py

```diff
@@ -124,14 +124,34 @@ def expand_image(image: str) -> str:
     """"base_image" entry must be a string OR dict with 'cpu' and 'gpu' keys, not '{}'"""
 )
 
-CalibanConfig = s.Schema({
-    s.Optional("build_time_credentials", default=False): bool,
-    s.Optional("default_mode", default=JobMode.CPU): s.Use(JobMode.parse),
+GCloudConfig = {
     s.Optional("project_id"): s.And(str, len),
-    s.Optional("cloud_key"): s.And(str, len),
-    s.Optional("base_image", default=None): BaseImage,
-    s.Optional("apt_packages", default=AptPackages.validate({})): AptPackages
-})
+    s.Optional("cloud_key"): s.And(str, len)
+}
+
+# Config items that are project-specific, and don't belong in a global
+# .calibanconfig shared between projects.
+ProjectConfig = {
+    s.Optional("build_time_credentials", default=False):
+        bool,
+    s.Optional("base_image", default=None):
+        BaseImage,
+    s.Optional("apt_packages", default=AptPackages.validate({})):
+        AptPackages,
+    # If present, Caliban will attempt to install Julia into the base container.
+    s.Optional("julia_version", default=None):
```

Comment:
This one's for you, @eschnett . I'll open up a separate PR that integrates your Julia work into `caliban.build` and pulls the version from this key. What do you think?